tests/platform_tests/api/test_sfp.py

```diff
@@ -259,6 +259,9 @@ def test_get_transceiver_info(self, duthosts, enum_rand_one_per_hwsku_hostname,
 
                     unexpected_keys = set(actual_keys) - set(self.EXPECTED_XCVR_INFO_KEYS + self.NEWLY_ADDED_XCVR_INFO_KEYS)
                     for key in unexpected_keys:
+                        #hardware_rev is applicable only for QSFP-DD
+                        if key == 'hardware_rev' and info_dict["type_abbrv_name"] == "QSFP-DD":
+                            continue
```

Comment:
@aravindmani-1 @prgeor - we are trying to run this test against QSFP28. Could you please elaborate on the reasoning with ignoring hardware_rev for a QSFP-DD only and not other SFP types as well.

Reply:
As i mentioned in the earlier comments that with the earlier SFP implementation, hardware_rev is used in place of vendor_rev key. hardware_rev is not applicable to media types except CMIS. From the unexpected key list, hardware_rev will be returned for QSFP-DD and the test case should not fail here. That's why i'm ignoring hardware_rev key for QSFP-DD.

Reply:
Based on the changes test_get_transceiver_info, the test is expecting that all vendors are using sfp-refactored API's beyond 202111 release.  

Nokia (and possibly other vendors) has not transitioned its implementation to this sfp-refactored API's. Instead, our code returns both 'hardware_rev' and 'vendor_rev' for all sfp types to be compatible with both earlier and later versions of xcvrd.

Would like to suggest to modify the test such that both hardware_rev and vendor_rev keys are acceptable to return for any module type. Regardless of the implementation of the xcvrd (earlier or later), xcvrd.py will just ignore the extraneous key.

```
                    for key in unexpected_keys:
                        if key == 'hardware_rev' or key == 'vendor_rev':
                            continue
                        self.expect(False, "Transceiver {} info contains unexpected field '{}'".format(i, key))
```

It  may be useful to note that Nokia's PMON code is under sub-module control,  whereas most vendors PMON code is directly under sonic-buildimage.

Reply:
It is expected for all the vendors to migrate to SFP refactored code. Is there any issue in migrating to SFP refactored code?. Could you clarify the return types used for hardware_rev and vendor_rev for all SFP types?. I don't see both keys present at the same time in any of the SFP specifications except CMIS.